### PR TITLE
Vcr

### PR DESCRIPTION
**Pull request by maelle**
Why use caching: make tests work in the absence of an internet connection; make them run faster in the presence of an internet connection.

Workflow used for this PR
* `vcr::use_vcr()`
* Add `vcr::use_cassette()` commands in tests
* Add a few more tests by looking at codecov report
* For new tests try to get in the habit of wrapping things involving http requests in `vcr::use_cassette()`
* Run the tests to create the cassettes/fixtures
* Turn off wifi and run the tests :slightly_smiling_face: 

```r
> sum(fs::file_size(fs::dir_ls("tests/fixtures/")))
257K
```

Not sure I'm doing this right, but the speed gain is very small in this case 

``` r
library("bench")
run_tests <- function(vcr_off,
                      pkg_path = "/home/maelle/Documents/R-hub/pkgsearch") {
  Sys.setenv("VCR_TURN_OFF" = vcr_off)
  devtools::test(pkg_path,
                 reporter = testthat::SilentReporter)

  return(TRUE)
}

bench::mark(
  iterations = 20,
  vcr_off = run_tests(TRUE),
  vcr_on = run_tests(FALSE)
)
#> Warning: Some expressions had a GC in every iteration; so filtering is disabled.
#> # A tibble: 2 x 6
#>   expression      min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr> <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 vcr_off       6.86s    7.69s     0.104    41.6MB    0.352
#> 2 vcr_on        5.26s    5.98s     0.154    32.1MB    1.10
```

<sup>Created on 2020-02-11 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ matrix:
    - r: 3.3
    - r: 3.4
    - r: 3.5
+     env: VCR_TURN_OFF=true
```
Comment (maelle) [inline .travis.yml]: It's useful I think to have one build with vcr turned off. That said it might be better to have it run regularly, maybe easier if this repo CI switches to GH Actions (where a workflow could be run every day or so).

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1 @@
+tests/fixtures/**/* -diff
```
Comment (maelle) [inline .gitattributes]: well it doesn't seem to hide fixtures from this PR diff :crying_cat_face:

File: tests/testthat/helper.r [hunk 3]
```diff
@@ -1,8 +1,4 @@
 
-skip_if_offline <- function() {
```
Comment (maelle) [inline tests/testthat/helper.r]: no longer needed if using cassettes

File: tests/testthat/test-public-api.r [hunk 4]
```diff
@@ -43,29 +46,62 @@ test_that("cran_events() works", {
...
   expect_equal(length(r4), 2)
   expect_equal(names(r4[[1]]), c("date", "name", "event", "package"))
 
 })
 
 test_that("cran_package_history() edge cases", {
 
-  skip_if_offline()
-
+  set.seed(42)
```
Comment (maelle) [inline tests/testthat/test-public-api.r]: maybe skip_if_offline and not removing the randomness would make more sense...

**Comment by maelle** [comment]
argh of course the Appveyor failure is due to encoding :grimacing:

File: tests/testthat/test-print.R [hunk 5]
```diff
@@ -2,25 +2,25 @@
...
-  x <- ps("csardi")
+  vcr::use_cassette("ps_csardi", {
+    x <- ps("csardi")
+  })
   out <- capture.output(print(x))
   expect_equal(length(out), 12)
   expect_match(out[1], "csardi.*packages in.*seconds")
   expect_match(out[2], "#\\s+package\\s+version\\s+by\\s+@\\s+title")
-  expect_match(out[3], "100\\s+igraph.*Gábor Csárdi.*Network Analysis")
+  expect_match(out[3], "100\\s+igraph.*Gábor Csárdi.*Network Analysis", useBytes = TRUE)
```
Comment (maelle) [inline tests/testthat/test-print.R]: this is the thing I needed to make the test pass on Appveyor, not sure why. :upside_down_face:

**Comment by gaborcsardi** [comment]
I decided to switch to curl, instead of httr, because curl is a bit lighter, and then I can use pkgsearch in pak. So I'll need to close this, sorry.
